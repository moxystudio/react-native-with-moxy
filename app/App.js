import 'react-native-gesture-handler';
import React from 'react';
import AppStack, { rootNavigation } from './navigation';
import { NavigationContainer } from '@react-navigation/native';

const App = () => (
    <NavigationContainer ref={ rootNavigation }>
        <AppStack />
    </NavigationContainer>
);

export default App;
