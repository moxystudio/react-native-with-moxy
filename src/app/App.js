import 'react-native-gesture-handler';
import React from 'react';
import { ThemeProvider } from '../shared/components';
import AppStack, { rootNavigation } from '../navigation';
import { NavigationContainer } from '@react-navigation/native';

const App = () => (
    <ThemeProvider>
        <NavigationContainer ref={ rootNavigation.navigationRef }>
            <AppStack />
        </NavigationContainer>
    </ThemeProvider>
);

export default App;
