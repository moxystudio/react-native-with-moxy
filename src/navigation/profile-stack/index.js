import React from 'react';
import { createStackNavigator } from '@react-navigation/stack';
import ProfileHeader from './header';
import ProfileScreen1 from '../../screens/profile/1/ProfileScreen1';
import ProfileScreen2 from '../../screens/profile/2/ProfileScreen2';

const Stack = createStackNavigator();

const ProfileStack = () => (
    <Stack.Navigator screenOptions={ {
        header: ProfileHeader,
    } }>
        <Stack.Screen name="Profile1" component={ ProfileScreen1 } />
        <Stack.Screen name="Profile2" component={ ProfileScreen2 } />
    </Stack.Navigator>
);

export default ProfileStack;
