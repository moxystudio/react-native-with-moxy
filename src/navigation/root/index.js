import * as React from 'react';

export const navigationRef = React.createRef();

export function navigate(name, params) {
    /* eslint-disable-next-line babel/no-unused-expressions */
    navigationRef.current?.navigate(name, params);
}
