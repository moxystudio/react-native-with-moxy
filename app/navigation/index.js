import React from 'react';
import * as rootNavigation from './root';
import { createStackNavigator } from '@react-navigation/stack';

import HomeScreen from '../screens/home/HomeScreen';

const Stack = createStackNavigator();

const AppStack = () => (
    <Stack.Navigator>
        <Stack.Screen name="Home" component={ HomeScreen } />
    </Stack.Navigator>
);

export default AppStack;

export {
    rootNavigation,
};
