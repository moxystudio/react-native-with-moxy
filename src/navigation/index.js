import React from 'react';
import * as rootNavigation from './root';
import { createStackNavigator } from '@react-navigation/stack';

import ProfileStack from './profile-stack';
import HomeScreen from '../screens/home/HomeScreen';

const Stack = createStackNavigator();

const AppStack = () => (
    <Stack.Navigator headerMode="screen">
        <Stack.Screen name="Home" component={ HomeScreen } />
        <Stack.Screen name="Profile" component={ ProfileStack } options={ { header: () => null } } />
    </Stack.Navigator>
);

export default AppStack;

export {
    rootNavigation,
};
