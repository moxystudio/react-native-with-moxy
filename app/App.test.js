import React from 'react';
import App from './App';
import { shallow, mount } from 'enzyme';
import { rootNavigation } from './navigation';

describe('app tests', () => {
    it('should render correctly', () => {
        const tree = shallow(<App />);

        expect(tree).toMatchSnapshot();
    });

    it('should have navigation ref', () => {
        mount(<App />);

        rootNavigation.navigate('Home');

        expect(rootNavigation).toEqual(expect.objectContaining({
            navigate: expect.any(Function),
        }));
    });
});
