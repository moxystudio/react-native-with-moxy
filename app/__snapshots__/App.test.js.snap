// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<ForwardRef(NavigationContainer)>
  <AppStack />
</ForwardRef(NavigationContainer)>
`;
